ImageViewer/Source/GalleryPagingDataSource.swift

```diff
@@ -16,7 +16,7 @@ final class GalleryPagingDataSource: NSObject, UIPageViewControllerDataSource {
 
     fileprivate let configuration: GalleryConfiguration
     fileprivate var pagingMode = GalleryPagingMode.standard
-    fileprivate let itemCount: Int
+    fileprivate var itemCount: Int { return itemsDataSource!.itemCount() }
```

Comment:
Absolutely agree with you, I try to follow a very similar approach when it comes to `!`
In this specific case I feel like it'd be reasonable to return `0` for `itemCount` in case the dataSource isn't set. How would `return itemsDataSource?.itemCount() ?? 0` make you feel? :)